aten/src/ATen/record_function.cpp

```diff
@@ -199,6 +199,23 @@ inline CallbackManager& manager() {
 
 } // namespace
 
+RecordFunctionCallbacks _getTLSCallbacks() {
+  return sorted_tls_callbacks_;
+}
+
+void _setTLSCallbacks(const RecordFunctionCallbacks& callbacks) {
+  clearThreadLocalCallbacks();
+  // keep the original handles
+  sorted_tls_callbacks_ = callbacks;
+  std::sort(
```

Comment:
is it just a precaution?

Reply:
yes, I assume this function is not perf sensitive and leaving it up to caller would be too fragile